dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java

```diff
@@ -3761,6 +3740,91 @@ private void synchWrite() throws IOException {
+     * if writing headers that the headers have been preprocessed and are ready to be written out
+     *
+     * @param finalWrite dictates if last http content should be written with trailers if any
+     */
+    private void nettyWrite(boolean sendHeaders, boolean finalWrite) throws IOException{
+        WsByteBuffer[] writeBuffers = getBuffList();
+        
+        if(!(getTSC().getWriteInterface() instanceof NettyTCPWriteRequestContext))
+            throw new RuntimeException("Writing on Netty requires a NettyTCPWriteRequestContext");
+
+        if (null != writeBuffers) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "Writing " + writeBuffers.length + " buffers on netty channel.");
+            }
+            
+            // If sendHeaders, we add the headers to the write interface to be written when data is written
+            if(sendHeaders) {
+                // Set prefix object on Netty Write Request Context
+                ((NettyTCPWriteRequestContext)getTSC().getWriteInterface()).queuePrefixObject(nettyResponse);
+            }
+
+            getTSC().getWriteInterface().setBuffers(writeBuffers);
+            try {
+                if(!nettyContext.channel().isOpen()){
+                    throw new IOException("Attempted to write on a closed Netty channel");
+                }
+                getTSC().getWriteInterface().write(TCPWriteRequestContext.WRITE_ALL_DATA, null, false, getWriteTimeout());
+            } finally {
+                // 457369 - disconnect write buffers in TCP when done
+                getTSC().getWriteInterface().setBuffers(null);
+            }
+
+        }
+        else if (sendHeaders) {
+            sendNettyHeaders();
+        }
+        else{
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "Netty write has no data to send.");
+            }
+        }
+
+        if(finalWrite) {
+            sendNettyFinalContent();
+        }
+    }
+
+    private void sendNettyFinalContent() {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "Netty write flushing out last http content due to final write happening.");
+        }
+        NettyResponseMessage resp = (NettyResponseMessage) getResponse();
+        HttpHeaders trailers = resp.getNettyTrailers();
+        DefaultLastHttpContent lastContent;
+        if (trailers.isEmpty())
+            lastContent = new LastStreamSpecificHttpContent(Integer.valueOf(nettyResponse.headers().get(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text(),
+                                                                                                        "-1")));
+        else {
+            lastContent = new LastStreamSpecificHttpContent(Integer.valueOf(nettyResponse.headers().get(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text(),
```

Comment:
It might be possible to only use the new constructor with the trailers since trailers will never be null and it being empty possibly doesnt change anything internally. Requires some testing.